### PR TITLE
SAR-3984 added stub v2 page

### PR DESCRIPTION
**Pull request by ChrisHillCap**

### CODE REVIEW

File: app/controllers/testonly/StubController.scala [hunk 1]
```diff
@@ -23,35 +22,85 @@ object TestSetupForm {
...
+    journeyconfigV2.copy(
+      options = journeyconfigV2.options.copy(
+        continueUrl = controllers.testonly.routes.StubController.showResultOfJourney(id).url)
+    )
+
+  val defaultJourneyConfigV2JsonAsString: JsValue = Json.parse(
+    """{
+      "version": 2,
+      "options":{
+      "continueUrl": "This will be ignored"
```
Comment (palequezal) [inline app/controllers/testonly/StubController.scala]: Would be better to build from the model and store as a constant, then convert to json here

File: it/controllers/StubControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    .bindings(bind[IdGenerationService].toInstance(MockIdGenerationService))
+    .configure(fakeConfig("application.router" -> "testOnlyDoNotUseInAppConf.Routes"))
+    .build
+
+  s"${controllers.testonly.routes.StubController.showStubPageForJourneyInitV2().url}" should {
+    "return 200" in {
+      val res = buildClientTestOnlyRoutes(path = "v2/test-setup")
+        .withHeaders(HeaderNames.COOKIE -> sessionCookieWithCSRF, "Csrf-Token" -> "nocheck")
+        .get()
+      await(res).status shouldBe 200
```
Comment (palequezal) [inline it/controllers/StubControllerISpec.scala]: Easier to read/more idiomatic to use play status constants. Plus, that's what we've done in the rest of the tests now so would make sense to be consistent.

File: it/controllers/StubControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            continueUrl = "testContinueUrl"
+          )
+      )))
+      val updatedContinueUrlForBasicJourney = Json.toJson(JourneyDataV2(JourneyConfigV2(
+        version = 2,
+        options = JourneyOptions(
+          continueUrl = s"/lookup-address/test-only/end-of-journey/$testJourneyId"
+        )
+      )))
+      val basicJourney =
```
Comment (palequezal) [inline it/controllers/StubControllerISpec.scala:49]: There's already a minimal config constant that could be used here

File: it/controllers/StubControllerISpec.scala [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      stubKeystoreSave(testJourneyId, basicJDataForBasicJourney, OK)
+      stubKeystoreSave(testJourneyId, updatedContinueUrlForBasicJourney, OK)
+
+      val res = buildClientTestOnlyRoutes(path = "v2/test-setup")
+        .withHeaders(HeaderNames.COOKIE -> sessionCookieWithCSRF, "Csrf-Token" -> "nocheck")
+        .post(Map(
+          "journeyConfig" -> Seq(basicJourney)))
+
+      val response = await(res)
+      response.status shouldBe 303
```
Comment (palequezal) [inline it/controllers/StubControllerISpec.scala]: Status constants

File: it/controllers/StubControllerISpec.scala (hunk 2 above)
Comment (ChrisHillCap) [reply it/controllers/StubControllerISpec.scala]: I agree with this one

File: it/controllers/StubControllerISpec.scala (hunk 4 above)
Comment (ChrisHillCap) [reply it/controllers/StubControllerISpec.scala]: I prefer the 303 but i have changed

File: it/controllers/StubControllerISpec.scala (hunk 3 above)
Comment (ChrisHillCap) [reply it/controllers/StubControllerISpec.scala:49]: I'd prefer to leave in the test for readability

**Pull request merged**